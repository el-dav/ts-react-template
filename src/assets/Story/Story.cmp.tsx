import React from 'react';

import Theme from '~/assets/Theme';
import Store from '~/assets/Store';

interface Props {
  isConnected?: boolean;
}

const Story: React.SFC<Props> = ({ children, isConnected }) =>
  isConnected ? (
    <Store>
      <Theme>{children}</Theme>
    </Store>
  ) : (
    <Theme>{children}</Theme>
  );

export default Story;
