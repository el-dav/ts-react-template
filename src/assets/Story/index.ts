import Story from './Story.cmp';

export default Story;
