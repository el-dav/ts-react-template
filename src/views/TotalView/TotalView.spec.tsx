import React from 'react';
import renderer from 'react-test-renderer';

import Story from 'assets/Story';

import TotalViewCmp from './TotalView.cmp';
import TotalViewCnt from './TotalView.cnt';

const increase = () => undefined;
const decrease = () => undefined;
const increaseAsync = () => undefined;

describe('TotalView', () => {
  it('renders the Component correctly', () => {
    const tree = renderer
      .create(
        <Story isConnected>
          <TotalViewCmp
            total={5}
            increase={increase}
            decrease={decrease}
            increaseAsync={increaseAsync}
          />
        </Story>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Component correctly with different counts', () => {
    const tree = renderer
      .create(
        <Story isConnected>
          <TotalViewCmp
            total={2}
            increase={increase}
            decrease={decrease}
            increaseAsync={increaseAsync}
          />
        </Story>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Container correctly', () => {
    const tree = renderer
      .create(
        <Story isConnected>
          <TotalViewCnt />
        </Story>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
