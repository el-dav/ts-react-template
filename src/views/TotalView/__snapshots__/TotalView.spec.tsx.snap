// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalView renders the Component correctly 1`] = `
<div
  className="css-9erftz"
>
  Total: 5
  <div
    className="css-rfmghh"
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Increase
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Decrease
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
  </div>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Increase Async
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
</div>
`;

exports[`TotalView renders the Component correctly with different counts 1`] = `
<div
  className="css-9erftz"
>
  Total: 2
  <div
    className="css-rfmghh"
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Increase
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Decrease
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
  </div>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Increase Async
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
</div>
`;

exports[`TotalView renders the Container correctly 1`] = `
<div
  className="css-9erftz"
>
  Total: 0
  <div
    className="css-rfmghh"
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Increase
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Decrease
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
  </div>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Increase Async
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
</div>
`;
